Fix bug #524300.  The upstream change was:

commit 7745384080ef70f7710530afa3e45477b126e056
Author: Zhang Rui <rui.zhang@intel.com>
Date:   Fri Oct 17 01:42:41 2008 -0400

    don't load asus-acpi if model is not supported

--- a/drivers/acpi/asus_acpi.c
+++ b/drivers/acpi/asus_acpi.c
@@ -1244,6 +1244,8 @@ static int asus_hotk_get_info(void)
 			       "default values\n", string);
 			printk(KERN_NOTICE
 			       "  send /proc/acpi/dsdt to the developers\n");
+			kfree(model);
+			return -ENODEV;
 		}
 		hotk->methods = &model_conf[hotk->model];
 		return AE_OK;
